internal/commands/app/create_inputs_test.go

```diff
@@ -0,0 +1,272 @@
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			var appFilter realm.AppFilter
+			rc := mock.RealmClient{}
+			rc.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				appFilter = filter
+				return []realm.App{testApp}, tc.findAppErr
+			}
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				tc.procedure(console)
+			}()
+
+			name, err := tc.inputs.resolveAppName(ui, rc, tc.from)
+
+			console.Tty().Close() // flush the writers
+			<-doneCh              // wait for procedure to complete
+
+			assert.Equal(t, tc.expectedErr, err)
+			assert.Equal(t, tc.expectedName, name)
+			assert.Equal(t, tc.expectedFilter, appFilter)
+		})
+	}
+}
+
+func TestAppCreateInputsResolveProject(t *testing.T) {
+	testApp := realm.App{
+		ID:          primitive.NewObjectID().Hex(),
+		GroupID:     primitive.NewObjectID().Hex(),
+		ClientAppID: "test-app-abcde",
+		Name:        "test-app",
+	}
+
+	for _, tc := range []struct {
+		description     string
+		inputs          createInputs
+		procedure       func(c *expect.Console)
+		groupsErr       error
+		expectedProject string
+		expectedErr     error
+	}{
+		{
+			description:     "should return project if project is set",
+			inputs:          createInputs{newAppInputs: newAppInputs{Project: testApp.GroupID}},
+			procedure:       func(c *expect.Console) {},
+			expectedProject: testApp.GroupID,
+		},
+		{
+			description: "should prompt for project if project is not set",
+			procedure: func(c *expect.Console) {
+				c.ExpectString("Atlas Project")
+				c.Send(testApp.GroupID)
+				c.SendLine(" ")
+				c.ExpectEOF()
+			},
+			expectedProject: testApp.GroupID,
+		},
+		{
+			description: "should error when finding group",
+			procedure:   func(c *expect.Console) {},
+			groupsErr:   errors.New("atlas client error"),
+			expectedErr: errors.New("atlas client error"),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			ac := mock.AtlasClient{}
+			ac.GroupsFn = func() ([]atlas.Group, error) {
+				return []atlas.Group{{ID: testApp.GroupID}}, tc.groupsErr
+			}
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				tc.procedure(console)
+			}()
+
+			projectID, err := tc.inputs.resolveProject(ui, ac)
+
+			console.Tty().Close() // flush the writers
+			<-doneCh              // wait for procedure to complete
+
+			assert.Equal(t, tc.expectedErr, err)
+			assert.Equal(t, tc.expectedProject, projectID)
+		})
+	}
+}
+
+func TestAppCreateInputsResolveDirectory(t *testing.T) {
+	t.Run("should return path of wd with app name appended", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
```

Comment:
we don't actually need a temporary directory created here, might be worth adding `mock.NewProfileFromWd` and have that use `os.Getwd` to set `WorkingDirectory` field.  no need for the `teardown` func either then